include/dfrn.php

```diff
@@ -0,0 +1,795 @@
+<?php
+require_once('include/items.php');
+require_once('include/Contact.php');
+require_once('include/ostatus.php');
+
+/**
+ * @brief Generates the atom entries for delivery.php
+ *
+ * This function is used whenever content is transmitted via DFRN.
+ *
+ * @param array $items Item elements
+ * @param array $owner Owner record
+ *
+ * @return string DFRN entries
+ */
+function dfrn_entries($items,$owner) {
+
+	$doc = new DOMDocument('1.0', 'utf-8');
+	$doc->formatOutput = true;
+
+	$root = dfrn_add_header($doc, $owner, "dfrn:owner", "", false);
+
+	if(! count($items))
+		return trim($doc->saveXML());
+
+	foreach($items as $item) {
+		$entry = dfrn_entry($doc, "text", $item, $owner, $item["entry:comment-allow"], $item["entry:cid"]);
+		$root->appendChild($entry);
+	}
+
+	return(trim($doc->saveXML()));
+}
+
+/**
+ * @brief Generate an atom feed for the given user
+ *
+ * This function is called when another server is pulling data from the user feed.
+ *
+ * @param App $a
+ * @param string $dfrn_id DFRN ID from the requesting party
+ * @param string $owner_nick Owner nick name
+ * @param string $last_update Date of the last update
+ * @param int $direction Can be -1, 0 or 1.
+ *
+ * @return string DFRN feed entries
+ */
+function dfrn_feed(&$a, $dfrn_id, $owner_nick, $last_update, $direction = 0) {
+
+	$sitefeed    = ((strlen($owner_nick)) ? false : true); // not yet implemented, need to rewrite huge chunks of following logic
+	$public_feed = (($dfrn_id) ? false : true);
+	$starred     = false;   // not yet implemented, possible security issues
+	$converse    = false;
+
+	if($public_feed && $a->argc > 2) {
+		for($x = 2; $x < $a->argc; $x++) {
```

Comment:
BTW: we now have the functions argc()  (https://github.com/friendica/friendica/blob/develop/boot.php#L2095) and argv($x) (https://github.com/friendica/friendica/blob/develop/boot.php#L2105) so you don't need to commit the whole app $a to a function anymore to get argv or argc